cli/reference/blocks/map.md

```diff
@@ -4,3 +4,128 @@ description: Learn how to define complex maps and objects inside globals and let
 ---
 
 # The `map` block
+
+Use the `map` block to convert a list into a map (key-value pair) or a nested map within a `globals` or `lets` block. 
```

Comment:
Can we explain the actual use case here too? What's a typical use case for the `map` block?